pkg/machinery/config/encoder/encoder.go

```diff
@@ -152,21 +172,21 @@ func toYamlNode(in interface{}, flags CommentsFlags) (*yaml.Node, error) {
 			}
 
 			var (
-				defined = isSet(v.Field(i))
+				empty = isEmpty(v.Field(i))
+				null  = isNil(v.Field(i))
 
 				skip   bool
 				inline bool
 				flow   bool
 			)
 
-			for i, part := range parts {
-				// always skip the first argument
-				if i == 0 {
-					continue
+			for _, part := range parts {
+				if part == "omitempty" && empty {
+					skip = true
 				}
 
-				if part == "omitempty" && !defined {
-					skip = true
+				if part == "omitonlyifnil" && !null {
```

Comment:
I liked `omitnil` more :))

Reply:
so the actual tag is `omitempty` + `omitonlyifnil`, so it kind of cancels the first one a bit